packages/api/src/controllers/v1/collection/mod.rs

```diff
@@ -30,70 +49,151 @@ pub struct CollectionControllerV1 {
 }
 
 impl CollectionControllerV1 {
-    pub fn new(pool: sqlx::PgPool) -> Self {
+    async fn query(
+        &self,
+        auth: Option<Authorization>,
+        param: CollectionQuery,
+    ) -> Result<QueryResponse<CollectionSummary>> {
+        if auth.is_none() {
+            return Err(ApiError::Unauthorized);
+        }
+
+        let mut total_count = 0;
+        let items: Vec<CollectionSummary> = CollectionSummary::query_builder()
+            .limit(param.size())
+            .page(param.page())
+            .query()
+            .map(|row: PgRow| {
+                use sqlx::Row;
+                total_count = row.try_get("total_count").unwrap_or_default();
+                row.into()
+            })
+            .fetch_all(&self.pool)
+            .await?;
+
+        Ok(QueryResponse { total_count, items })
+    }
+
+    async fn create(
+        &self,
+        auth: Option<Authorization>,
+        agit_id: i64,
+        CollectionCreateRequest { title }: CollectionCreateRequest,
+    ) -> Result<Json<Collection>> {
+        if auth.is_none() {
+            return Err(ApiError::Unauthorized);
+        }
+
+        let collection = self.repo.insert(agit_id, title).await?;
+        Ok(Json(collection))
+    }
+
+    async fn update(
+        &self,
+        auth: Option<Authorization>,
+        id: i64,
+        param: CollectionUpdateRequest,
+    ) -> Result<Collection> {
+        if auth.is_none() {
+            return Err(ApiError::Unauthorized);
+        }
+
+        let collection = self.repo.update(id, param.into()).await?;
+        Ok(collection)
+    }
+
+    async fn delete(&self, id: i64, auth: Option<Authorization>) -> Result<Collection> {
+        if auth.is_none() {
+            return Err(ApiError::Unauthorized);
+        }
+
+        self.repo.delete(id).await.map_err(|_| ApiError::NotFound)
+    }
+}
+
+impl CollectionControllerV1 {
+    pub fn new(pool: sqlx::Pool<sqlx::Postgres>) -> Self {
         let repo = Collection::get_repository(pool.clone());
         Self { repo, pool }
     }
 
     pub fn route(pool: sqlx::PgPool) -> Result<by_axum::axum::Router> {
         let ctrl = Self::new(pool);
         Ok(by_axum::axum::Router::new()
-            .route("/:id", post(Self::act_by_id).get(Self::get))
-            .route("/", post(Self::act).get(Self::list))
+            .route(
+                "/:id",
+                get(Self::get_collection_by_id).post(Self::act_collection_by_id),
+            )
+            .route("/", post(Self::act_collection).get(Self::get_collection))
+            .route("/create/:agit_id", post(Self::act_collection))
```

Comment:
@dotandev why did you add this route? I think it is same with 'post /' right?

Reply:
@dotandev then I think this part should be move to under agit controller. 

It is much suitable to `POST /v1/agits/:agit-id/collections`